### PR TITLE
boards/nz32-sc151: add initial support

### PR DESCRIPTION
**Pull request by aabadie**
This PR supersedes #6081 and is based on ~~#6819~~ (merged).

### CODE REVIEW

**Comment by aabadie** [comment]
@fjmolinas, I think this PR is ready. I fixed a few things in the board config:
* extra SPI,
* only one I2C
* 3 UARTS
* 1 PWM

If you have access to the hardware one day, feel free to test. Maybe @kYc0o can test as well.

**Comment by aabadie** [comment]
Needs #6819 first

**Comment by kYc0o** [comment]
Why?

**Comment by aabadie** [comment]
Because this PR is based on it

**Comment by aabadie** [comment]
Well, I think I see your point but I find it a bit overkill. Of course, I can make this PR being mergeable on its own without dependency to ADC and open another PR adding ADC to this board. I think it makes not so useful extra work to achieve the same result.

**Comment by kYc0o** [comment]
It's OK, let's merge #6819 first.

**Comment by aabadie** [comment]
rebased

**Comment by aabadie** [comment]
#6819 has been merged.

**Comment by aabadie** [comment]
Rebased and checked again.
Tested periph ADC, PWM, UART, RTC, Timer and xtimer all are working. Anyone for a review ?

**Comment by aabadie** [comment]
Nobody to review/test this ? (this is more than ready)

File: boards/nz32-sc151/Makefile.include [hunk 1]
```diff
@@ -0,0 +1,25 @@
+## the cpu to build for
+export CPU = stm32l1
+export CPU_MODEL = stm32l151rc
+
+# define the default port depending on the host OS
+PORT_LINUX ?= /dev/ttyUSB0
+PORT_DARWIN ?= $(firstword $(sort $(wildcard /dev/tty.SLAB_USBtoUART*)))
```
Comment (kYc0o) [inline boards/nz32-sc151/Makefile.include]: Can @astralien3000 check if this works?

File: boards/nz32-sc151/Makefile.include [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+# set the default id
+export ID ?= 0483:df11
+
+export BINFILE = $(patsubst %.elf,%.bin,$(ELFFILE))
+
+export FLASHER = dfu-util
+export DEBUGGER = # spark core has no debugger
```
Comment (kYc0o) [inline boards/nz32-sc151/Makefile.include]: spark core? And AFAIK, we can still use a JLINK debugger for this board...

Comment (aabadie) [inline boards/nz32-sc151/Makefile.include]: probably copy pasted, will remove this line

Comment (aabadie) [inline boards/nz32-sc151/Makefile.include]: Well as it's configured by default for using dfu-util, this variable should be kept empty (but maybe I'm wrong)

**Comment by aabadie** [comment]
Another ready one waiting for an ACK: @kYc0o ?

**Comment by kYc0o** [comment]
Well, I still can't make it work on OS X, I did the trick with the buttons but the device doesn't appear in my /dev/ list...

**Comment by aabadie** [comment]
Can you try to keep the BOOT button pressed, plug the board and then release the button ?

**Comment by kYc0o** [comment]
I tried that without success... I'll ask someone with linux to try...

**Comment by aabadie** [comment]
@kYc0o could find someone to test ?

File: boards/nz32-sc151/include/periph_conf.h [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+        .miso_pin = GPIO_PIN(PORT_B, 14),
+        .sclk_pin = GPIO_PIN(PORT_B, 13),
+        .cs_pin   = GPIO_UNDEF,
+        .af       = GPIO_AF5,
+        .rccmask  = RCC_APB1ENR_SPI2EN,
+        .apbbus   = APB1
+    },
+    {
+        .dev      = SPI3,
+        .mosi_pin = GPIO_PIN(PORT_C, 12),
```
Comment (fjmolinas) [inline boards/nz32-sc151/include/periph_conf.h]: ADC and SPI3 share same ports, therefore PC10-12 are doubly assigned.

File: boards/nz32-sc151/include/periph_conf.h [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+
+/**
+ * @name    ADC configuration
+ * @{
+ */
+#define ADC_CONFIG {              \
+    { GPIO_PIN(PORT_C, 0), 10 },  \
+    { GPIO_PIN(PORT_C, 1), 11 },  \
+    { GPIO_PIN(PORT_C, 2), 12 },  \
+    /* ADC Temperature channel */ \
```
Comment (fjmolinas) [inline boards/nz32-sc151/include/periph_conf.h]: I would suggest PC15 is also enabled since it is used for the board battery measuring circuit.

**Comment by fjmolinas** [comment]
Works on linux, ran tests for Timer, RTC, UART, SPI & ADC. Will test on OSX.

**Comment by aabadie** [comment]
> Works and tested on linux and OSX. Ran tests for Timer, RTC, UART, SPI & ADC.

Thanks for testing @fjmolinas ! Maybe the extra ADC (on PC15) could be added in a follow-up PR.

**Comment by kYc0o** [comment]
Ok well let's go with this one after a second Murdock test.

**Comment by kYc0o** [comment]
Everything OK, so go!

**Pull request merged**